controllers/userController.js

```diff
@@ -1,43 +1,158 @@
-import { users } from "../models";
+import { users, drivers, operators, buses } from "../models";
 import { success,fail,sendError } from "../function/respond.js";
-import db from "../models/index.js";
+import sendMail from '../utils/sendEmail'
+import { validateUsersOnCreate, validateDriverId } from "../function/validation.js"
+import generator from 'generate-password'
+import bcrypt from 'bcrypt'
 
+const passwordNew = generator.generate({
+	length: 10,
+	numbers: true
+});
 
 const getAllUsers = async (req, res) => {
- 
-		/* ======= Start:: List all users with count ========== */ 
-			// users.findAll().then(users => {
-			// 	return success(res,200,users,"Retrieved");
-			// })
-		/* ======= End:: List all users with count ============ */ 
-		
-	
-	
-		/* ======= Start:: List all users =================== */ 
-			users.findAndCountAll().then(users => {
-				
-				return success(res,200,users,"Retrieved");
+			users.findAll({where: {isDeleted: false}, attributes:{exclude: ['password','createdAt', 'updatedAt', 'roles', 'isDeleted']}}).then(user => {
+				return success(res,200,user,"Retrieved", req);
 			})
 		/* ========= End:: List all users ================== */ 
 	
 };
 
 const createUser = async (req, res) => {
-    try {
+		
+		const hashedPassword = await bcrypt.hash(passwordNew, 10)
 
 	    /* =============================== start: Validation ============================== */ 
-			if(!req.body.username && !req.body.password &&  !req.body.fullname && !req.body.email) throw new Error('Body is required');				
-			if(!req.body.password || req.body.password.trim() === "") return fail(res,400,req.body,"Please make sure you add password");
-		/* ================================= End: Validation ============================== */ 
-		
+		const { firstname, lastname, username, email, telephone, userType } = req.body
+		const { error } = validateUsersOnCreate({firstname:firstname, lastname:lastname, username:username, email:email, telephone:telephone, userType:userType})
+		if(error) { 
+			return fail(res, 400, error.details[0].message)
+		}
+		const userExist = await users.findAll({
```

Comment:
Just use findByPk or findOne, they return the first entry when found rather than findAll